doc/source/dev_guide/aux_data.rst

```diff
@@ -0,0 +1,122 @@
+
+        def __init__(self, name, url=None, known_hash=None, **kwargs):
+            super().__init__(name, **kwargs)
+            data_files = [{'url': url, 'known_hash': known_hash}]
+            self.register_data_files(data_files)
+
+However your code registers files, to be consistent it must do it during
+initialization so that the :func:`~satpy.data_download.find_registerable_files`.
+If your component isn't a reader, writer, or compositor then this function
+will need to be updated to find and load your registered files. See
+:ref:`offline_aux_downloads` below for more information.
+
+As mentioned, the mixin class is included in the base reader and writer class.
+To register files in these cases, include a ``data_files`` section in your
+YAML configuration file. For readers this would go under the ``reader``
+section and for writers the ``writer`` section. This parameter is a list
+of dictionaries including a ``url``, ``known_hash``, and optional
+``filename``. For example::
+
+    reader:
+        name: abi_l1b
+        short_name: ABI L1b
+        long_name: GOES-R ABI Level 1b
+        ... other metadata ...
+        data_files:
+          - url: "https://example.com/my_data_file.dat"
+          - url: "https://raw.githubusercontent.com/pytroll/satpy/master/README.rst"
+            known_hash: "sha256:5891286b63e7745de08c4b0ac204ad44cfdb9ab770309debaba90308305fa759"
+          - url: "https://raw.githubusercontent.com/pytroll/satpy/master/RELEASING.md"
+            filename: "satpy_releasing.md"
+            known_hash: null
+
+See the :class:`~satpy.data_download.DataDownloadMixin` for more information.
+
+Retrieving
+^^^^^^^^^^
+
+Files that have been registered (see above) can be retrieved by calling the
+:func:`~satpy.data_download.retrieve` function. This function expects a single
+argument: the cache key. Cache keys are returned by registering functions, but
+can also be pre-determined by following the scheme
+``<component_type>/<filename>`` (ex. ``readers/README.rst``).
+Retrieving a file will download it to local disk if needed and then return
+the local pathname. Data is stored locally in the :ref:`data_dir_setting`.
+It is up to the caller to then open the file.
+
+.. _offline_aux_downloads:
+
+Offline Downloads
+-----------------
+
+To assist with operational environments, Satpy includes a
+:func:`~satpy.data_download.retrieve_all` function that will try to find all
+files that Satpy components may need to download in the future and download
+them to the current directory specified by :ref:`data_dir_setting`.
+This function allows you to specify a list of ``readers``, ``writers``, or
+``composite_sensors`` to limit what components are checked for files to
+download.
+
+The ``retrieve_all`` function is also available through a command line script
+called ``satpy_retrieve_all``. Run the following for usage information.
+
+.. code-block:: bash
+
+    satpy_retrieve_all --help
```

Comment:
I'm feeling `satpy_retrieve_all` can be a bit misleading for an inexperienced user, how about `satpy_retrieve_all_static_data`?

Reply:
What about `satpy_retrieve_all_aux_data` considering that's the naming in other places? Maybe I should rename `satpy/data_download.py` to `satpy/aux_download.py`? Or `aux_data.py`?

Reply:
`satpy_retrieve_all_aux_data` sounds good. I like `aux_download.py`, but the current name is fine by me.